Feature/363 관리자 회원 상태 변경 및 차단 기능 구현, 권한 없는 요청 응답 메시지 적용

higukang opened the pull request:
# 🚀 Pull Request
 관리자 회원 상태 변경 및 차단 기능 구현, 권한 없는 요청 응답 메시지 적용

## #️⃣ 연관된 이슈

#363 

## 📋 작업 내용

- 회원 status 목록에 DELETED 추가
- Member.java : 멤버 상태 (Status) 비활성화, 삭제 상태로 바꿀 수 있도록 변경
- MemberService.java : 삭제되거나 비활성화된 회원 재 활성화 기능 추가
- AdminMemberController.java : 회원 재활성화 기능 추가
- CustomUserDetailsService.java : 삭제되거나 비활성화된 회원 로그인 못하도록 조건 추가
- Dto 파일 스웨거 예시 변경
- MemberService.getMemberStatus : 회원의 Id를 받아 상태를 리턴하는 메서드 추가
- JWTFilter.doFilterInternal : 요청 엑세스 토큰의 회원 Id 를 받아 DB에 해당 회원 상태조회 후 활성화된 회원인 지 확인 하는 조건 추가
- ErrorCode.INACTIVE_OR_DELETED_MEMBER : 에러코드 추가
- CustomAccessDeniedHandler에서 BaseResponse를 사용하여 응답 메시지 통일
- ErrorCode Enum에 FORBIDDEN(403) 메시지 수정
- Unauthorized 요청 시 클라이언트가 일관된 형식으로 응답을 처리할 수 있도록 개선

higukang committed:
chore : 회원 status 목록에 DELETED 추가

#Issue: 363

higukang committed:
Merge remote-tracking branch 'origin/feature/363-회원-상태status-변경-기능-구현-및-회원-상태에-삭제-추가' into feature/363-회원-상태status-변경-기능-구현-및-회원-상태에-삭제-추가

higukang committed:
feat :회원 재활성화 기능 추가

- Member.java : 멤버 상태 (Status) 비활성화, 삭제 상태로 바꿀 수 있도록 변경
- MemberService.java : 삭제되거나 비활성화된 회원 재 활성화 기능 추가
- AdminMemberController.java : 회원 재활성화 기능 추가
- CustomUserDetailsService.java : 삭제되거나 비활성화된 회원 로그인 못하도록 조건 추가
- Dto 파일 스웨거 예시 변경

#Issue: 363

higukang committed:
feat: 회원 비활성화 처리 기능 구현

- Member.deactivateAccount : 함수명 변경
- MemberService.deactivateMember : 회원 상태를 INACTIVE로 바꾸는 비활성화 기능 추가
- AdminMemberController.deactivateMember : 회원 상태를 INACTIVE로 바꾸는 비활성화 기능 추가

#Issue: 363

oowtl reviewed (approved):
회원 삭제 및 비활성화 처리 코드 확인하였습니다.

higukang committed:
feat: 관리자 회원 차단 기능 구현

- MemberService.getMemberStatus : 회원의 Id를 받아 상태를 리턴하는 메서드 추가
- JWTFilter.doFilterInternal : 요청 엑세스 토큰의 회원 Id 를 받아 DB에 해당 회원 상태조회 후 활성화된 회원인 지 확인 하는 조건 추가
- ErrorCode.INACTIVE_OR_DELETED_MEMBER : 에러코드 추가

#Issue: 363

higukang committed:
feat: 권한 없는 요청에 대한 응답 메시지 적용

- CustomAccessDeniedHandler에서 BaseResponse를 사용하여 응답 메시지 통일
- ErrorCode Enum에 FORBIDDEN(403) 메시지 수정
- Unauthorized 요청 시 클라이언트가 일관된 형식으로 응답을 처리할 수 있도록 개선

#Issue: 363
